pallets/phala-world/src/incubation.rs

```diff
@@ -115,7 +164,34 @@ pub mod pallet {
 			nft_id: NftId,
 		) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
-			// TODO: Move to incubation.rs
+			// Check if IncubationPhase is enabled
+			ensure!(CanStartIncubation::<T>::get(), Error::<T>::StartIncubationNotAvailable);
+			// Ensure that the collection is an Origin of Shell Collection
+			ensure!(
+				Self::is_origin_of_shell_collection_id(collection_id),
+				Error::<T>::WrongCollectionId
+			);
+			// Ensure sender is owner
+			ensure!(Self::is_owner(sender.clone(), collection_id, nft_id), Error::<T>::NotOwner);
```

Comment:
```suggestion
			ensure!(Self::is_owner(&sender, collection_id, nft_id), Error::<T>::NotOwner);
```

Not super important but better to pass by ref. AccountId32 is a small array